src/app/modules/form/public-form/public-form.middlewares.ts

```diff
@@ -1,9 +1,54 @@
 import { RequestHandler } from 'express'
 import { StatusCodes } from 'http-status-codes'
+import mongoose from 'mongoose'
 
+import { createLoggerWithLabel } from '../../../../config/logger'
 import { WithForm } from '../../../../types'
+import getSubmissionModel from '../../../models/submission.server.model'
 import { FormDeletedError } from '../form.errors'
-import { isFormPublic } from '../form.service'
+import { deactivateForm, isFormPublic } from '../form.service'
+
+const SubmissionModel = getSubmissionModel(mongoose)
+
+const logger = createLoggerWithLabel(module)
+
+/**
+ * Express middleware function that checks if a form has exceeded its submission limits before allowing
+ * downstream middleware to handle the request. Otherwise, it returns a HTTP 404.
+ */
+export const checkFormSubmissionLimitAndDeactivate: RequestHandler = (
+  req,
+  res,
+  next,
+) => {
+  const { form } = req as WithForm<typeof req>
+  if (form.hasSubmissionLimit) {
+    return SubmissionModel.countDocuments({ form: form._id })
+      .exec()
+      .then(async (queryResults) => {
```

Comment:
can convert this part to async/await too?

Reply:
done